plugins/acpi-dmar/meson.build

```diff
@@ -1,6 +1,4 @@
-if get_option('hsi') and \
-   host_machine.system() == 'linux' and \
-   (host_cpu == 'x86' or host_cpu == 'x86_64')
+if hsi and (host_cpu == 'x86' or host_cpu == 'x86_64')
```

Comment:
the reason for this pattern in addition to at top level `meson.build` is so that if/when HSI is expanded to other architectures we make sure that X86 specific plugins don't compile for those architectures.